### ISSUE TITLE
AppCompat 24 issue

### ISSUE THREAD

**Issue by khrizt** [Yalantis/uCrop#137]
It seems like there's a bug with AppCompat 24.0.0 and Google Play Services 9 [http://stackoverflow.com/a/37337203](Google employee telling so), the error stack trace is:

```java
FATAL EXCEPTION: IntentService[RegIntentService]
   Process: [...], PID: 10798
   Theme: themes:{}
   java.lang.IncompatibleClassChangeError: The method 'java.io.File android.support.v4.content.ContextCompat.getNoBackupFilesDir(android.content.Context)' was expected to be of type virtual but instead was found to be of type direct (declaration of 'com.google.android.gms.iid.zzd' appears in /data/app/[...]-1/base.apk)
       at com.google.android.gms.iid.zzd.zzeb(Unknown Source)
       at com.google.android.gms.iid.zzd.<init>(Unknown Source)
       at com.google.android.gms.iid.zzd.<init>(Unknown Source)
       at com.google.android.gms.iid.InstanceID.zza(Unknown Source)
       at com.google.android.gms.iid.InstanceID.getInstance(Unknown Source)
       at package.services.RegistrationIntentService.onHandleIntent(RegistrationIntentService.java:43)
       at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:66)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:148)
       at android.os.HandlerThread.run(HandlerThread.java:61)
```

I've downloaded the library and downgrade to SDK 23 and AppCompat 23.4.0 and it works perfectly, I've made a pull request ([#136](https://github.com/Yalantis/uCrop/pull/136)) for these changes but I opened this issue to discuss the consequences of this change.

**Comment by khrizt** [comment]
Did you had some time to take a look at this?

**Comment by shliama** [comment]
Hi :) I think it says:

"update may 27:

we just released an update (version 9.0.1) to fix the incompatibility I mentioned in my first edit.
Please update your dependencies and let us know if this is still an issue.

Thanks!"

So why should I downgrade sdk version?

**Comment by khrizt** [comment]
I think this is an incompatibility with using SDK 23 and AppCompat 24 (as uCrop is using). I tried upgrading to google-services 9 and failed anyway. I'll try to gather more data to fully understand the error combination and update this issue.
